app/recurring_job.go

```diff
@@ -375,7 +413,34 @@ func (job *Job) doSnapshot() (err error) {
 	return nil
 }
 
+func (job *Job) deleteSnapIfRetainNumExceeded(volume *longhornclient.Volume) error {
+	if job.task == longhorn.RecurringJobTypeSnapshotForceCreate {
```

Comment:
Suggest moving this to the caller to reduce this method's complexity.

Reply:
Moved the condition check to the caller. Thanks